packages/aws-cdk/bin/cdk.ts

```diff
@@ -37,6 +37,7 @@ async function parseCommandLineArguments() {
     .option('proxy', { type: 'string', desc: 'Use the indicated proxy. Will read from HTTPS_PROXY environment variable if not specified.', requiresArg: true })
     .option('ca-bundle-path', { type: 'string', desc: 'Path to CA certificate to use when validating HTTPS requests. Will read from AWS_CA_BUNDLE environment variable if not specified.', requiresArg: true })
     .option('ec2creds', { type: 'boolean', alias: 'i', default: undefined, desc: 'Force trying to fetch EC2 instance credentials. Default: guess EC2 instance status.' })
+    .option('endpoints', { type: 'string', default: undefined, desc: 'Overwrite service endpoints in AWS SDK.' })
```

Comment:
What is the syntax here? Should be documented. It seems like this is  json map, but feels like it won’t be super comfortable... is this how the CLI takes it? How do people know which endpoints they need to provide?

Alternatively you can maybe use this syntax:

```
—endpoint-s3=URL
```

Reply:
FYI, localstack has deprecated individual endpoints, see https://github.com/localstack/localstack

```
Starting with version 0.11.0, all APIs are exposed via a single edge service,
which is accessible on http://localhost:4566 by default ... please reconfigure
your client SDKs to start using the single edge endpoint URL!
```

Reply:
If we did that it would mean this could only be used for localstack. In which case we might as well call it `--localstack-address`.

Reply:
somehow didn't get a GitHub notification for updates to this thread.

@eladb Agreed, I need to add it to the README, so the syntax is something like:

```
cdk deploy --endpoints 'cloudformation=http://localhost:4581,s3=http://localhost:4572'
cdk deploy --endpoints '{"cloudformation":"http://localhost:4581","s3":"http://localhost:4572"}'
cdk deploy --endpoints $(cat endpoints.json)
```

The format of the endpoints.json file should be:
```
{
    "cloudformation": "http://localhost:4581",
    "s3": "http://localhost:4572"
}
```

@rix0rrr I kind of agree that we might just want to support a single endpoint at the outset and it overrides all endpoints. I'm more reluctant to tie the parameter name to localstack though as I'd still want to stay more generic than that.

let me know what you think? If we're in favour of a single endpoint, I'll write tests accordingly

Reply:
Well.. For me, back when i created this, it was purely for localstack. That said i do agree with @shivlaks in not tying the naming to localstack, and trying to keep it generic. 
It is a bit tough though, as i think the new single-endpoint server in localstack is indigenous to localstack only.

Reply:
That's what I'm trying to say. A single endpoint for all services only makes sense for localstack.

Guess there's nothing wrong with allowing a single URL to apply to all services... but let's all be very clear it's for exactly one use case :)